cloudflare/resource_cloudflare_load_balancer.go

```diff
@@ -0,0 +1,335 @@
+		newLoadBalancer.RegionPools = expandGeoPools(regionPools, "region")
+	}
+
+	if popPools, ok := d.GetOk("pop_pools"); ok {
+		newLoadBalancer.PopPools = expandGeoPools(popPools, "pop")
+	}
+
+	zoneName := d.Get("zone").(string)
+	zoneId, err := client.ZoneIDByName(zoneName)
+	if err != nil {
+		return fmt.Errorf("error finding zone %q: %s", zoneName, err)
+	}
+	d.Set("zone_id", zoneId)
+
+	log.Printf("[INFO] Creating CloudFlare Load Balancer from struct: %+v", newLoadBalancer)
+
+	r, err := client.CreateLoadBalancer(zoneId, newLoadBalancer)
+	if err != nil {
+		return errors.Wrap(err, "error creating load balancer for zone")
+	}
+
+	if r.ID == "" {
+		return fmt.Errorf("cailed to find id in Create response; resource was empty")
+	}
+
+	d.SetId(r.ID)
+
+	log.Printf("[INFO] CloudFlare Load Balancer ID: %s", d.Id())
+
+	return resourceCloudFlareLoadBalancerRead(d, meta)
+}
+
+func resourceCloudFlareLoadBalancerUpdate(d *schema.ResourceData, meta interface{}) error {
+	// since api only supports replace, update looks a lot like create...
+	client := meta.(*cloudflare.API)
+	zoneId := d.Get("zone_id").(string)
+
+	loadBalancer := cloudflare.LoadBalancer{
+		ID:           d.Id(),
+		Name:         d.Get("name").(string),
+		FallbackPool: d.Get("fallback_pool_id").(string),
+		DefaultPools: expandInterfaceToStringList(d.Get("default_pool_ids")),
+		Proxied:      d.Get("proxied").(bool),
+		TTL:          d.Get("ttl").(int),
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		loadBalancer.Description = description.(string)
+	}
+
+	if regionPools, ok := d.GetOk("region_pools"); ok {
+		loadBalancer.RegionPools = expandGeoPools(regionPools, "region")
+	}
+
+	if popPools, ok := d.GetOk("pop_pools"); ok {
+		loadBalancer.PopPools = expandGeoPools(popPools, "pop")
+	}
+
+	log.Printf("[INFO] Updating CloudFlare Load Balancer from struct: %+v", loadBalancer)
+
+	_, err := client.ModifyLoadBalancer(zoneId, loadBalancer)
+	if err != nil {
+		return errors.Wrap(err, "error creating load balancer for zone")
+	}
+
+	return resourceCloudFlareLoadBalancerRead(d, meta)
+}
+
+func expandGeoPools(pool interface{}, geoType string) map[string][]string {
+	cfg := pool.(*schema.Set).List()
+	expanded := make(map[string][]string)
+	for _, v := range cfg {
+		locationConfig := v.(map[string]interface{})
+		// assume for now that lists are of type interface{} by default
+		expanded[locationConfig[geoType].(string)] = expandInterfaceToStringList(locationConfig["pool_ids"])
+	}
+	return expanded
+}
+
+func resourceCloudFlareLoadBalancerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneId := d.Get("zone_id").(string)
+	loadBalancerId := d.Id()
+
+	loadBalancer, err := client.LoadBalancerDetails(zoneId, loadBalancerId)
+	if err != nil {
+		if strings.Contains(err.Error(), "HTTP status 404") {
+			log.Printf("[INFO] Load balancer %s in zone %s no longer exists", loadBalancerId, zoneId)
+			d.SetId("")
+			return nil
+		} else {
```

Comment:
Nitpick/Go idiom/**not a blocker:** the `} else { ` is not needed because the statement above returns, the `else` returns, and the containing block ends.

The following are logical equivalent:

```go
{
  if true {
    return true
  } else {
    return false
  }
}
```

and

```go
{
  if true {
    return true
  }
  return false
}
```